api/execute_transactions.go

```diff
@@ -215,8 +215,29 @@ func (c *Client) ProcessTransactions(dataDir string, processType string) error {
 	return nil
 }
 
+// extractPresidentNameFromPath extracts the president's name from the file path.
+// It expects the path to contain either "/orgchart/PresidentName/" or "/people/PresidentName/".
+func extractPresidentNameFromPath(filePath string) (string, error) {
+	pathParts := strings.Split(filepath.ToSlash(filePath), "/")
+	for i, part := range pathParts {
+		if part == "orgchart" || part == "people" || part == "documents" {
```

Comment:
Just a thought you can do this later, make sure these constant values are not used in the code as strings when they are required, make sure they are constant values or enums defined in some part of the code.